src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (!herb.hidden && herb instanceof Herbivore) {
+      herb.health -= 50;
+      Animal.alive = Animal.alive.filter((el) => el.health !== 0);
```

Comment:
Instead of filtering every time a bite happens, consider checking the health of the herbivore immediately after it is bitten and removing it from the `Animal.alive` array if its health is less than or equal to 0. This avoids unnecessary iterations over the `Animal.alive` array.